src/main/kotlin/pt/up/fe/ni/website/backend/controller/PostController.kt

```diff
@@ -32,4 +32,7 @@ class PostController(private val service: PostService) {
 
     @DeleteMapping("/{postId}")
     fun deletePost(@PathVariable postId: Long) = service.deletePostById(postId)
+
+    @GetMapping("/{postSlug}**")
+    fun getPost(@PathVariable postSlug: String) = service.getPostBySlug(postSlug)
```

Comment:
Also a small detail, but I'd personally put this get next to id get, so we can see them next to each other and rapidly understand why the `\\d´` and `**` modifiers are used